client/packages/invoices/src/InboundShipment/DetailView/columns.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  ArrayUtils,
+  getLinesFromRow,
+  usePreferences,
+  useTranslation,
+  ColumnDef,
+  Groupable,
+  ColumnType,
+} from '@openmsupply-client/common';
+import { useInboundShipmentLineErrorContext } from '../context/inboundShipmentLineError';
+import { useMemo } from 'react';
+import { isInboundPlaceholderRow } from '../../utils';
+import { InboundLineFragment } from '../api';
+
+export const useInboundShipmentColumns = () => {
+  const t = useTranslation();
+  const {
+    manageVaccinesInDoses,
+    allowTrackingOfStockByDonor,
+    manageVvmStatusForStock,
+  } = usePreferences();
+  const { getError } = useInboundShipmentLineErrorContext();
+
+  return useMemo((): ColumnDef<Groupable<InboundLineFragment>>[] => {
+    return [
+      {
+        accessorKey: 'comment',
+        header: t('label.comment'),
+        columnType: ColumnType.Comment,
+      },
+      {
+        accessorKey: 'item.code',
+        header: t('label.code'),
+        size: 120,
+        pin: 'left',
+        enableColumnFilter: true,
+        enableSorting: true,
+        getIsError: row =>
+          getLinesFromRow(row).some(
+            r => getError(r)?.__typename === 'LineLinkedToTransferredInvoice'
+          ),
+      },
```

Comment:
It's pinned, but may be slightly better to put this first in the column array anyway?